app/controllers/precheck_emails_controller.rb

```diff
@@ -0,0 +1,46 @@
+# TODO: Mark as pre checked complete on confirm action
+
+class PrecheckEmailsController < ApplicationController
+  before_action :find_token, only: %i[new confirm reject]
+
+  def new
+    render 'error' and return unless @token
```

Comment:
Would it make sense to extract this to a method?

Reply:
I originally tried to do that, but it doesn't seem to work. Controllers are a bit funny with the return.

Have you done something like this before by chance? Any suggestions on how to?

Reply:
Oh yes I see what you mean now, one way to handle that is to raise an error and use `rescue_from`

I think it would be fine to leave it how it is though.